packages/core/test/acceptance/profiler_spec.ts

```diff
@@ -0,0 +1,242 @@
+
+  afterAll(() => setProfiler(null));
+
+  describe('change detection hooks', () => {
+    it('should call the profiler for creation and change detection', () => {
+      @Component({selector: 'my-comp', template: '<button (click)="onClick()"></button>'})
+      class MyComponent {
+        onClick() {}
+      }
+
+      TestBed.configureTestingModule({declarations: [MyComponent]});
+      const fixture = TestBed.createComponent(MyComponent);
+
+      expect(profilerSpy).toHaveBeenCalled();
+
+      expect(profilerSpy.calls.argsFor(0)).toEqual([
+        ProfilerEvent.CreationStart, fixture.componentInstance, undefined
+      ]);
+      expect(profilerSpy.calls.argsFor(1)).toEqual([
+        ProfilerEvent.CreationEnd, fixture.componentInstance, undefined
+      ]);
+
+      fixture.detectChanges();
+
+      const [changeDetectionStart, changeDetectionEnd] = profilerSpy.calls.all().slice(-2);
+
+      expect(changeDetectionStart.args).toEqual([
+        ProfilerEvent.ChangeDetectionStart, fixture.componentInstance, undefined
+      ]);
+      expect(changeDetectionEnd.args).toEqual([
+        ProfilerEvent.ChangeDetectionEnd, fixture.componentInstance, undefined
+      ]);
+    });
+
+    it('should invoke the profiler when the template throws', () => {
+      @Component({selector: 'my-comp', template: '{{ throw() }}'})
+      class MyComponent {
+        throw() {
+          throw new Error();
+        }
+      }
+
+      TestBed.configureTestingModule({declarations: [MyComponent]});
+
+      let myComp: MyComponent;
+      expect(() => {
+        const fixture = TestBed.createComponent(MyComponent);
+        myComp = fixture.componentInstance;
+        fixture.detectChanges();
+      }).toThrow();
+
+      expect(profilerSpy).toHaveBeenCalled();
+
+      expect(profilerSpy.calls.argsFor(0)).toEqual([
+        ProfilerEvent.CreationStart, myComp!, undefined
+      ]);
+      expect(profilerSpy.calls.argsFor(1)).toEqual([ProfilerEvent.CreationEnd, myComp!, undefined]);
+    });
+  });
+
+  describe('outputs and events', () => {
+    it('should invoke the profiler on event handler', () => {
+      @Component({selector: 'my-comp', template: '<button (click)="onClick()"></button>'})
+      class MyComponent {
+        onClick() {}
+      }
+
+      TestBed.configureTestingModule({declarations: [MyComponent]});
+      const fixture = TestBed.createComponent(MyComponent);
+      const myComp = fixture.componentInstance;
+
+      const clickSpy = spyOn(myComp, 'onClick');
+      const calls = profilerSpy.calls.all();
+      const button = fixture.nativeElement.querySelector('button');
+
+      button?.click();
+
+      expect(clickSpy).toHaveBeenCalled();
+
+      const outputStart = calls.map((call: any) => call.args)
+                              .find((arg: any) => arg[0] === ProfilerEvent.OutputStart);
+      const outputEnd = calls.map((call: any) => call.args)
+                            .find((arg: any) => arg[0] === ProfilerEvent.OutputEnd);
```

Comment:
I'd propose creating small util function:

```
function findProfilerCall(eventType: ProfilerEvent): any[] {
  return profilerSpy.calls.all().map((call: any) => call.args).find((arg: any) => arg[0] === eventType);
}
```
and use it in all test cases like:
```suggestion
      const outputStart = findProfilerCall(ProfilerEvent.OutputStart);
      const outputEnd = findProfilerCall(ProfilerEvent.OutputEnd);
```